Add nav redesign frame to Reader /read and /discover behind flag

candy02058912 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/dotcom-forge/issues/6856

## Proposed Changes

* Adds the frame to /read and /discover
* Still behind the flag because we don't want to make things even more inconsistent with some parts of the Reader with the frame and some don't

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Test /read, /discover and see if there's anything weird
* Disable the flag `?flags=-layout/dotcom-nav-redesign-v2` and see if there are regressions


https://github.com/Automattic/wp-calypso/assets/6586048/f2e11e04-b288-4989-babc-c23689d83945




https://github.com/Automattic/wp-calypso/assets/6586048/50b52b33-03b8-458b-8277-3c4be0df7d48





## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

candy02058912 committed:
Add nav redesign frame to Reader Recent and Discover behind flag

davemart-in commented:
Nice @candy02058912! I noticed that responsive web is showing an excessive amount of padding now:

![CleanShot 2024-05-03 at 09 00 37@2x](https://github.com/Automattic/wp-calypso/assets/5634774/6963233a-01ed-4126-a2f4-e24f363e6b4b)

dsas commented:
One small thing I noticed is that the stream items in /discover are narrower than the ones in /read. This means it is showing less of the excerpt than /read and production /discover.

https://github.com/Automattic/wp-calypso/assets/93301/9ac1c6e3-cda7-4d71-9f1f-35e194cff7e3

candy02058912 committed:
Update styles for /read and /discover

candy02058912 committed:
Fix styles

candy02058912 commented:
Appreciate the catches!
I've also added videos for /read and /discover in the PR description.

> Nice @candy02058912! I noticed that responsive web is showing an excessive amount of padding now:

fixed ✅ 

> One small thing I noticed is that the stream items in /discover are narrower than the ones in /read. This means it is showing less of the excerpt than /read and production /discover.

fixed ✅

miksansegundo reviewed (approved):
LGTM 👍 The issues from previous reviews are fixed, and I haven't noticed any issues with the flag or any regresions without it.

**With flag**

https://github.com/Automattic/wp-calypso/assets/1881481/f6566b44-8de4-42d2-a860-8273856adfef

**Without flag**

https://github.com/Automattic/wp-calypso/assets/1881481/eb3677ed-937b-4e81-8dc1-fabe3478caa2
